pkg/sql/rename_index.go

```diff
@@ -36,6 +36,10 @@ type renameIndexNode struct {
 func (p *planner) RenameIndex(ctx context.Context, n *tree.RenameIndex) (planNode, error) {
 	_, tableDesc, err := expandMutableIndexName(ctx, p, n.Index, true /* requireTable */)
```

Comment:
Could you accomplish the same thing by passing requireTable=false here?

Reply:
I actually ended up making this more similar to drop_index.go, and set `requireTable=!n.IfExists`